core/vm/instructions.go

```diff
@@ -370,106 +370,107 @@ func opSAR(pc *uint64, evm *EVM, contract *Contract, memory *Memory, stack *Stac
 	return nil, nil
 }
 
-func opSha3(pc *uint64, evm *EVM, contract *Contract, memory *Memory, stack *Stack) ([]byte, error) {
+func opSha3(pc *uint64, interpreter *InterpreterEVM, contract *Contract, memory *Memory, stack *Stack) ([]byte, error) {
 	offset, size := stack.pop(), stack.pop()
 	data := memory.Get(offset.Int64(), size.Int64())
 	hash := crypto.Keccak256(data)
+	evm := interpreter.evm
```

Comment:
Are you using this anywhere? Seems like an unused variable? Does this code compile? ;P

Reply:
Ah, you did use it to add the preimages. Please use it to access the vmConfig too.